MantidQt/MantidWidgets/inc/MantidQtMantidWidgets/WorkspacePresenter/ADSNotifiable.h

```diff
@@ -0,0 +1,53 @@
+#ifndef MANTID_MANTIDWIDGETS_ADSNOTIFIABLE_H_
+#define MANTID_MANTIDWIDGETS_ADSNOTIFIABLE_H_
+
+/**
+\class  ADSNotifiable
+\brief  Mixin class allows ADS notifications
+\author Lamar Moore
+\date   24-08-2016
+\version 1.0
+
+
+Copyright &copy; 2016 ISIS Rutherford Appleton Laboratory, NScD Oak Ridge
+National Laboratory & European Spallation Source
+
+This file is part of Mantid.
+
+Mantid is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 3 of the License, or
+(at your option) any later version.
+
+Mantid is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+File change history is stored at: <https://github.com/mantidproject/mantid>
+*/
+namespace MantidQt {
+namespace MantidWidgets {
+
+class ADSNotifiable {
+public:
+  virtual ~ADSNotifiable() {}
+
+  enum class Flag {
+    WorkspaceLoaded,
+	WorkspaceRenamed,
+    WorkspaceDeleted,
+    WorkspacesGrouped,
+    WorkspacesSorted,
```

Comment:
You've  put quite a lot of code/options in place prior to them being needed. Does an `ADSNotifiable` need to know about sorting, for example yet? And what tests cover this behaviour in the PR. The structured does look good thus far, but please don't extend it any further until we start coding real test scenarios in, which we should make pass one-by-one.